src/cli/commands/install.js

```diff
@@ -234,14 +237,20 @@ export class Install {
     // exclude package names that are in install args
     const excludeNames = [];
     for (const pattern of excludePatterns) {
-      // can't extract a package name from this
-      if (getExoticResolver(pattern)) {
+      if (GitResolver.isVersion(pattern)) {
+        const pathname = urlParse(pattern).pathname;
+
+        if (pathname) {
+          const name = util.removeSuffix(pathname.split('/')[2], '.git');
+          excludeNames.push(name);
+        }
+      } else if (getExoticResolver(pattern)) {
```

Comment:
Can you instead try using [guessName util](https://github.com/yarnpkg/yarn/blob/8918da15c36e793c8f5ed7be24197a7b5afc89d4/src/util/guess-name.js), so something like
```
if (getExoticResolver(pattern)) {
   excludeNames.push(guessName(pattern))
}
```
This way we avoid git-specific logic in the install command file and other exotic resolvers and hosted git resolvers can't take advantage of this fix too :)